src/errors.rs

```diff
@@ -1,67 +1,211 @@
-use output;
-use std::ffi::OsString;
+use std::ffi;
+use std::fmt;
+use std::io;
 
-const ERROR_PREFIX: &'static str = "CLI assertion failed";
+use failure;
 
-fn format_cmd(cmd: &[OsString]) -> String {
+fn format_cmd(cmd: &[ffi::OsString]) -> String {
     let result: Vec<String> = cmd.iter()
         .map(|s| s.to_string_lossy().into_owned())
         .collect();
     result.join(" ")
 }
 
-error_chain! {
-    links {
-        Output(output::Error, output::ErrorKind);
+pub trait ChainFail {
+    fn chain<E>(self, cause: E) -> Self
```

Comment:
Needed an a nice way to chain without `failure::Error`.

With `failure::Error` not being a `Fail`, a developer basically needs to accept a `failure::Error` to be a universal recipient.  Otherwise its back to `enum` causes that are annoying to maintain.